tests/devices/test_default_mixed_autograd.py

```diff
@@ -0,0 +1,585 @@
+                [
+                    np.sin(theta) * np.cos(lam) * np.cos(phi),
+                    np.cos(theta) * np.cos(lam) * np.sin(phi) + np.sin(lam) * np.cos(phi),
+                    np.cos(theta) * np.sin(lam) * np.cos(phi) + np.cos(lam) * np.sin(phi),
+                ]
+            )
+            * 2
+            * (np.sin(lam) * np.sin(phi) - np.cos(theta) * np.cos(lam) * np.cos(phi))
+        )
+        assert np.allclose(res, expected_grad, atol=tol, rtol=0)
+
+    @pytest.mark.parametrize("interface", ["tf", "torch"])
+    def test_error_backprop_wrong_interface(self, interface, tol):
+        """Tests that an error is raised if diff_method='backprop' but not using
+        the Autograd interface"""
+        dev = qml.device("default.mixed", wires=1)
+
+        def circuit(x, w=None):
+            qml.RZ(x, wires=w)
+            return qml.expval(qml.PauliX(w))
+
+        msg = "Device default.mixed only supports diff_method='backprop' when using the autograd interface"
+        msg = re.escape(msg)
+        with pytest.raises(qml.QuantumFunctionError, match=msg):
+            qml.qnode(dev, diff_method="backprop", interface=interface)(circuit)
+
+    @pytest.mark.parametrize(
+        "dev_name,diff_method,mode",
+        [
+            ["default.mixed", "finite-diff", "backward"],
+            ["default.mixed", "parameter-shift", "backward"],
+            ["default.mixed", "backprop", "forward"],
+        ],
+    )
+    def test_ragged_differentiation(self, dev_name, diff_method, mode, tol):
+        """Tests correct output shape and evaluation for a tape
+        with prob and expval outputs"""
+        dev = qml.device(dev_name, wires=2)
+        x = np.array(0.543, requires_grad=True)
+        y = np.array(-0.654, requires_grad=True)
+
+        @qml.qnode(dev, diff_method=diff_method, interface="autograd", mode=mode)
+        def circuit(x, y):
+            qml.RX(x, wires=[0])
+            qml.RY(y, wires=[1])
+            qml.CNOT(wires=[0, 1])
+            return [qml.expval(qml.PauliZ(0)), qml.probs(wires=[1])]
+
+        res = circuit(x, y)
+
+        expected = np.array(
+            [np.cos(x), (1 + np.cos(x) * np.cos(y)) / 2, (1 - np.cos(x) * np.cos(y)) / 2]
+        )
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+        res = qml.jacobian(circuit)(x, y)
+        assert isinstance(res, tuple) and len(res) == 2
+        assert res[0].shape == (3,)
+        assert res[1].shape == (3,)
+
+        expected = (
+            np.array([-np.sin(x), -np.sin(x) * np.cos(y) / 2, np.sin(x) * np.cos(y) / 2]),
+            np.array([0, -np.cos(x) * np.sin(y) / 2, np.cos(x) * np.sin(y) / 2]),
+        )
+        assert np.allclose(res[0], expected[0], atol=tol, rtol=0)
+        assert np.allclose(res[1], expected[1], atol=tol, rtol=0)
+
+
+@pytest.mark.autograd
+class TestHighLevelIntegration:
+    """Tests for integration with higher level components of PennyLane."""
+
+    @pytest.mark.skip
```

Comment:
How come this is being skipped?

Reply:
The test is for whether the Hamiltonian is being split into its different terms when computing `expval`. This also seems out of scope of this PR, so I'll just remove it.

Reply:
Removed this test.